odbc/src/api/connect_attr_tests.rs

```diff
@@ -205,17 +214,38 @@ mod unit {
         }
     }
 
-    const UNSUPPORTED_ATTRS: [ConnectionAttribute; 18] = [
+    // Test setting the current catalog attribute.
+    #[test]
+    fn set_autocommit() {
+        unsafe {
+            let conn = Connection::with_state(std::ptr::null_mut(), ConnectionState::Connected);
+            let mongo_handle: *mut _ = &mut MongoHandle::Connection(conn);
+
+            let autocommit_value: UInteger = 0u32;
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLSetConnectAttrW(
+                    mongo_handle as *mut _,
+                    ConnectionAttribute::SQL_ATTR_AUTOCOMMIT as i32,
+                    autocommit_value as Pointer,
+                    0
+                )
+            );
+            let conn_handle = (*mongo_handle).as_connection().unwrap();
+            let attributes = &conn_handle.attributes.read().unwrap();
+            assert_eq!(attributes.autocommit, Autocommit::Off);
+        }
+    }
+
+    const UNSUPPORTED_ATTRS: [ConnectionAttribute; 15] = [
```

Comment:
Matt's JIRA ticket implies that when setting SQL_QUIET_MODE to a value, the driver would panic. This is odd. 
I didn't see a `set_unsupported_attr` test when I looked at the code previously, but it looks like this is what `set_invalid_attr` tests and that test is passing fine.

Could you please:
- Verify that the driver doesn't panic but report SQL_ERROR if you try to set for example `SQL_ATTR_TRACE` to `SQL_OPT_TRACE_ON`
- Rename the test `set_invalid_attr` -> `set_unsupported_attr`
- Add a test with an incorrect attribute identifier to also make sure that we report SQL_ERROR and don't crash (I'm pretty sure we don't, but I could see a validation)

Reply:
> Matt's JIRA ticket implies that when setting SQL_QUIET_MODE to a value, the driver would panic. This is odd.
I didn't see a set_unsupported_attr test when I looked at the code previously, but it looks like this is what set_invalid_attr tests and that test is passing fine.

Isn't this because Jon added this:
```
ConnectionAttribute::SQL_ATTR_APP_WCHAR_TYPE
| ConnectionAttribute::SQL_ATTR_QUIET_MODE => SqlReturn::SUCCESS,
```

Reply:
That was done after the ticket was created, so I don't think so, but I can't tell for sure 🤷‍♀️